src/basilisp/lang/tagged.py

```diff
@@ -0,0 +1,85 @@
+from typing import (
+    Optional,
+    TypeVar,
+)
+
+from typing_extensions import Unpack
+
+from basilisp.lang.interfaces import (
+    ILispObject,
+    ILookup,
+)
+
+from basilisp.lang.keyword import Keyword
+from basilisp.lang.obj import PrintSettings, lrepr
+from basilisp.lang.symbol import Symbol
+
+K = TypeVar("K")
+V = TypeVar("V")
+
+class TaggedLiteral(
+    ILispObject,
+    ILookup[K, V],
+):
+    """Basilisp TaggedLiteral. https://clojure.org/reference/reader#tagged_literals
+    """
+
+    __slots__ = ("_tag", "_form", "_hash")
+
+    def __init__(
+        self, tag: Symbol, form
+    ) -> None:
+        self._tag = tag
+        self._form = form
+        self._hash = -1
```

Comment:
Python hashes can be any integer (including negative values), so this should probably just be `None`.

Reply:
I updated this line to also include the type annotation, not sure if this is the right place to do so.

Reply:
Yep this is where you'd add a type annotation for an instance member.